include/numerics/dense_matrix.h

```diff
@@ -1015,17 +1015,17 @@ Real DenseMatrix<T>::l1_norm () const
 
 template<typename T>
 inline
-Real DenseMatrix<T>::linfty_norm () const
+auto DenseMatrix<T>::linfty_norm () const -> decltype(std::abs(T(0)))
 {
   libmesh_assert (this->_m);
   libmesh_assert (this->_n);
 
-  Real rowsum = 0.;
+  decltype(std::abs(T(0))) rowsum = 0.;
```

Comment:
Just curious, does `auto rowsum = std::abs(T(0));` work to initialize this, or do you really need `decltype` here?

Reply:
Yeah, I'm pretty sure that would work. It looks obvious now, but I went through so many iteration with the decltype arguments that initially I didn't think of that. I'll make the change.